src/pkrd/rng/mt.rs

```diff
@@ -0,0 +1,77 @@
+pub struct MT {
+    index: usize,
+    mt: [u32; 624],
+}
+
+impl MT {
+    pub fn new(seed: u32) -> MT {
+        let mut rng = MT {
+            mt: [0; 624],
+            index: 624,
+        };
+        rng.init(seed);
+
+        rng
+    }
+
+    fn init(&mut self, seed: u32) {
+        self.mt[0] = seed;
+
+        let mut seed = seed;
+        for i in 1..624 {
+            seed = 0x6c078965u32.wrapping_mul(seed ^ (seed >> 30)).wrapping_add(i);
+            self.mt[i as usize] = seed;
+        }
+    }
+
+    pub fn next(&mut self) -> u32 {
```

Comment:
What are thoughts on implementing rust's [Iterator trait](https://doc.rust-lang.org/std/iter/index.html)?

Implementing Iterator will give us free methods like `skip`, `take`, `find`, and `map`.  Then we can do things like `rng.skip(10).next()`, which seems like a common method we've implemented before.

```rust
impl iter::Iterator for MT {
    type Item = u32;

    fn next(&mut self) -> Option<Self::Item> {
        if self.index == 624 {
            self.shuffle();
        }

        let y = self.mt[self.index];
        self.index += 1;

        // Game doesn't store tempered in memory
        //y ^= y >> 11;
        //y ^= (y << 7) & 0x9d2c5680;
        //y ^= (y << 15) & 0xefc60000;
        //y ^= y >> 18;
        Some(y)
    }
}
```

Reply:
Iterator looks interesting but I question how it would be useful for us. All we are gonna do is check in a loop how many times frame has advanced. Would iterator make this any better?

Reply:
Good point.  The times we've had a `skip` type method have all been for desktop tools, but our uses for this will probably be fairly limited.

We can hold off for now and add it later if needed.